src/xwayland.c

```diff
@@ -7,6 +7,43 @@
 #include "view.h"
 #include "workspaces.h"
 
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
```

Comment:
I think we really need a generic `MIN(a, b)` and `MAX(a, b)` macro somewhere, its only a matter of time unless we get a redefinition error somewhere. I usually tend to define it before using and then undefine it right after usage but that produces some pretty ugly code.

Reply:
That would be nice. I would say just define MIN and MAX in labwc.h and be done with it.

Reply:
Have been thinking the same.
MAX/MIM are defined by cairo/pango, so we’d have to call it something else. 
Lower-case?

Reply:
This works fine (labwc.h):
```
#define MIN(a, b) (((a) < (b)) ? (a) : (b))
#define MAX(a, b) (((a) > (b)) ? (a) : (b))
```

However, this breaks (labwc.h, wrapped in unnecessary `(` and `)`):
```
#define MIN(a, b) ((((a) < (b)) ? (a) : (b)))
#define MAX(a, b) ((((a) > (b)) ? (a) : (b)))
```
```
In file included from ../src/osd.c:14:
../include/labwc.h:64: warning: "MIN" redefined
   64 | #define MIN(a, b) ((((a) < (b)) ? (a) : (b)))
      |
In file included from /usr/lib/aarch64-linux-gnu/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/pango-1.0/pango/pango-coverage.h:25,
                 from /usr/include/pango-1.0/pango/pango-font.h:25,
                 from /usr/include/pango-1.0/pango/pango-attributes.h:25,
                 from /usr/include/pango-1.0/pango/pango.h:25,
                 from /usr/include/pango-1.0/pango/pangocairo.h:25,
                 from ../src/osd.c:6:
/usr/include/glib-2.0/glib/gmacros.h:887: note: this is the location of the previous definition
```

So it seems to work when defining the macros exactly like in `gmacros.h`.
I don't know if we should rely on that though. We could also just include the glib macros instead.

Reply:
`#ifndef MIN/MAX` would also be fine by me. No strong preferences for either `#undef` or `#ifndef`.